src/validation.cpp

```diff
@@ -2270,13 +2270,30 @@ bool CChainState::FlushStateToDisk(
         CoinsCacheSizeState cache_state = GetCoinsCacheSizeState();
         LOCK(m_blockman.cs_LastBlockFile);
         if (fPruneMode && (m_blockman.m_check_for_pruning || nManualPruneHeight > 0) && !fReindex) {
-            // make sure we don't prune above the blockfilterindexes bestblocks
+            // make sure we don't prune above any of the prune locks bestblocks
             // pruning is height-based
             int last_prune = m_chain.Height(); // last height we can prune
+            std::optional<std::string> limiting_lock; // prune lock that actually was the limiting factor, only used for logging
+
             ForEachBlockFilterIndex([&](BlockFilterIndex& index) {
-               last_prune = std::max(1, std::min(last_prune, index.GetSummary().best_block_height));
+                last_prune = std::max(1, std::min(last_prune, index.GetSummary().best_block_height));
             });
 
+            for (const auto& prune_lock : m_blockman.m_prune_locks) {
+                assert(prune_lock.second.height_first < std::numeric_limits<int>::max());
+                // if (prune_lock.second.height_first == std::numeric_limits<int>::max()) continue;
```

Comment:
In commit "blockstorage: Add prune locks to BlockManager" (a473db7e6a97289f981c3ff4b560e07a680efed5)

Could drop the commented line

Reply:
🤦‍♂️  done